core/src/main/java/io/grpc/transport/WritableBufferAllocator.java

```diff
@@ -32,13 +32,14 @@
 package io.grpc.transport;
 
 /**
- * The preferred way of creating a {@link WritableBuffer}.
+ * An allocator of buffers provided by the transport implementation to {@link MessageFramer} so
+ * it can send chunks of data to the transport in a form that the transport can directly serialize.
  */
 public interface WritableBufferAllocator {
 
   /**
-   * Allocate a new {@link WritableBuffer} that can hold
-   * {@code capacity} bytes.
+   * Request a new {@link WritableBuffer} with the given {@code capacity}. The allocator is
+   * free to return a buffer with a greater or lesser capacity.
```

Comment:
Yes, it isn't done in this PR. Yes, `NettyWritableBufferAllocator` would be modified to be aware of the maximum _pooled_ buffer size.

This doesn't actually have anything to do with HTTP/2's frame sizes (for Netty; for OkHttp we are using it for that purpose). Instead, once the allocation is too large Netty will no longer pool the Buffer, which likely outweighs the cost of larger buffers.

My initial problem with this PR was that it seemed we would always make sure to use a single buffer able to hold the entire message. Being able to "max out" the requested size at 256K or 1M is the primary reason I'm okay with this PR now.